molecularnodes/entities/molecule/base.py

```diff
@@ -156,171 +192,130 @@ def n_models(self):
+        except AttributeError:
+            self.object["chain_ids"] = None
+
+        try:
+            self.object.mn.biological_assemblies = json.dumps(self.assemblies())  # type: ignore
+        except InvalidFileError:
+            self.object.mn.biological_assemblies = ""  # type: ignore
+
+    def _create_object(
+        self,
+        name="NewObject",
+    ) -> None:
+        """
+        Create a 3D model of the molecule, with one vertex for each atom.
 
         Parameters
         ----------
-        name : str, optional
-            The name of the model. Default is 'NewMolecule'.
-        style : str, optional
-            The style of the model. Default is 'spheres'.
-        selection : np.ndarray, optional
-            The selection of atoms to include in the model. Default is None.
-        build_assembly : bool, optional
-            Whether to build the biological assembly. Default is False.
-        centre : str, optional
-            Denote method used to determine center of structure. Default is '',
-            resulting in no translational motion being removed. Accepted values
-            are `centroid` or `mass`. Any other value will result in default
-            behavior.
-        del_solvent : bool, optional
-            Whether to delete solvent molecules. Default is True.
-        del_hydrogen: bool, optional
-            Whether to delete hydrogen atoms. Default is False.
-        collection : str, optional
-            The collection to add the model to. Default is None.
         verbose : bool, optional
             Whether to print verbose output. Default is False.
-        color : Optional[str], optional
-            The color scheme to use for the model. Default is 'common'.
 
         Returns
         -------
         bpy.types.Object
             The created 3D model, as an object in the 3D scene.
         """
-        is_stack = isinstance(self.array, struc.AtomArrayStack)
+        if self._code:
+            name = f"MN_{self._code}"
 
-        if isinstance(selection, np.ndarray):
-            array = self.array[selection]
+        if isinstance(self.atom_array, AtomArrayStack):
+            is_stack = True
+            array = self.atom_array[0]
         else:
-            array = self.array
-
-        # remove the solvent from the structure if requested
-        if del_solvent:
-            mask = np.invert(struc.filter_solvent(array))
-            if is_stack:
-                array = array[:, mask]
-            else:
-                array = array[mask]
-
-        if del_hydrogen:
-            mask = array.element != "H"
-            if is_stack:
-                array = array[:, mask]
-            else:
-                array = array[mask]
-
-        obj, frames = _create_object(
-            array=array,
+            is_stack = False
+            array = self.atom_array
+
+        self.object = databpy.create_object(
```

Comment:
I'd like to reiterate how much easier this is to follow the MN logic.  :thumbs-up: